perception/docker_tf/README.md

```diff
@@ -0,0 +1,264 @@
+
+### cvt_pascal.py
+
+This script downloads all of the images and xml files using the json script from labelbox to gather all of our data onto the container. It then puts this data into the PascalVOC format for easy processing. It creates two folders for these files.
+
+### split_data.py
+
+This script splits the xmls and images into a 80/20 split. 80% of the data goes to training, 20% to validation. It creates two folders and places the images into these. 
+
+### xml_to_csv.py
+
+Takes the xml files and puts them into the CSV format for generate_tfrecord.py
+
+### generate_tfrecord.py
+
+Takes the CSV files and image/label directories and generates the two tfrecords we need for the trainer. Currently you have to manually edit this file to add the classes you need. In the future this should change to be automated based on the labelmap.pbtxt file you also have to manually edit. This does make a good place to change what specific segments of your dataset you want to train on however. For example if my labelled data had 25 different classes labelled, but I only cared about one, I could only process the labels that have te correct name. An example of editing this file is as follows. Please note that the strings used must **EXACTLY** match the labels used in labelbox.
+
+If I wanted to go from training a network to find roulette_wheels and instead find the entire NaviGator course, I would have to change the way the tfrecord interprets the labels. The example change would be the following:
+
+#### Initial
+```python
+def class_text_to_int(row_label):
+    if row_label == 'Wheel':
+        return 1
+    else:
+        None
+```
+#### Modified
+
+```python
+def class_text_to_int(row_label):
+    if row_label == 'Red Totem':
+        return 1
+    elif row_label == 'Blue Totem':
+        return 2
+    elif row_label == 'Green Totem':
+        return 3
+    elif row_label == 'Yellow Totem':
+        return 4
+    elif row_label == 'Black Totem':
+        return 5
+    elif row_label == 'Black Buoy':
+        return 6
+    else:
+        None
+```
+
+### labelmap.pbtxt
+
+Currently you have to manually edit this file as well, which is unfortunate. Ideally only one of these files should be manually edited. This contains all the labels and matching ID's you will be using for your identification. This is found in the data directory, which is likely one directory above the one you are currently in (you should see a data folder, a labelbox_data_processing folder, and a train.sh script. You'll also need to make a models folder if one is not present there and place a model in it that you want to retrain). An example of this file is the following:
```

Comment:
Is this still accurate? Be sure to go through README and make sure it _all_ reflects the latest changes